File: newRoadSystem.cpp
bool newRoadSystem(std::vector<std::vector<bool>> roadRegister) {
    std::vector<int> v(roadRegister.size());
    for (int i = v.size(); i--;) {
        for (int j = v.size(); j--;) {
            if (roadRegister[i][j]) {
                v[i]++;
                v[j]--;
            }
        }
    }
    return std::all_of(v.begin(), v.end(), [](int n) {
        return n == 0;
    });
}

File: newRoadSystem.js
function newRoadSystem(roadRegister) {
    var cities = new Array(roadRegister.length);
    for(var i = 0; i < cities.length; i++) {
        cities[i] = new City(roadRegister[i]);
    }
    
    cities.forEach(function(city) {
       for(var i = 0; i < city.register.length; i++) {
           if(city.register[i]) {
               city.out++;
               cities[i].in++;
           }
       } 
    });
    
    for(var i = 0; i < cities.length; i++) {
        if(cities[i].in != cities[i].out) {
            return false;
        }
    }
    
    return true;
}

function City(register) {
    this.register = register;
    this.in = 0;
    this.out = 0;
}
